hazelcast-client/src/test/java/com/hazelcast/client/map/AbstractMapQueryPartitionIteratorTest.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.map;
+
+import com.hazelcast.client.proxy.ClientMapProxy;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.TruePredicate;
+import com.hazelcast.test.HazelcastTestSupport;
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NoSuchElementException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public abstract class AbstractMapQueryPartitionIteratorTest extends HazelcastTestSupport {
+
+    protected TestHazelcastFactory factory;
+    protected HazelcastInstance server;
+    protected HazelcastInstance client;
+
+    @After
+    public void teardown() {
+        factory.terminateAll();
+    }
+
+
+    @Test(expected = NoSuchElementException.class)
+    public void test_next_Throws_Exception_On_EmptyPartition() throws Exception {
+        final ClientMapProxy<String, String> proxy = getMapProxy();
```

Comment:
I like `final` and I add them to my code - so I should add them where missing. Shall we debate on this? :)